burnman/perplex.py

```diff
@@ -0,0 +1,294 @@
+# This file is part of BurnMan - a thermoelastic and thermodynamic toolkit for the Earth and Planetary Sciences
+# Copyright (C) 2012 - 2015 by the BurnMan team, released under the GNU
+# GPL v2 or later.
+
+from __future__ import absolute_import
+from __future__ import print_function
+import warnings
+
+
+from subprocess import Popen, PIPE, STDOUT
+from os import rename
+
+
+import numpy as np
+from scipy.interpolate import interp2d 
+
+from .material import Material, material_property
+from . import eos
+from .tools import copy_documentation
+
+def create_perplex_table(werami_path, project_name, outfile, n_pressures, n_temperatures, pressure_range=None, temperature_range=None):
+    '''  
+    This function uses PerpleX's werami software to output a table file containing the following material properties.            
+    2 - Density (kg/m3)                           
+    4 - Expansivity (1/K, for volume)                               
+    5 - Compressibility (1/bar, for volume)                     
+    10 - Adiabatic bulk modulus (bar)                                
+    11 - Adiabatic shear modulus (bar)                               
+    12 - Sound velocity (km/s)                                       
+    13 - P-wave velocity (Vp, km/s)                                  
+    14 - S-wave velocity (Vs, km/s)                               
+    17 - Entropy (J/K/kg)                                            
+    18 - Enthalpy (J/kg)                                             
+    19 - Heat Capacity (J/K/kg)                              
+    22 - Molar Volume (J/bar)        
+    '''
+    
+    print('Working on creating {0}x{1} P-T table file using werami. Please wait.\n'.format(n_pressures, n_temperatures))
+
+    try:
+        str2 = 'y\n{0} {1}\n{2} {3}\n'.format(pressure_range[0]/1.e5, pressure_range[1]/1.e5,
+                                              temperature_range[0], temperature_range[1])
+    except:
+        print('Keeping P-T range the same as the original project range.\n')
+        str2 = 'n\n'
+    
+    stdin='{0:s}\n2\n' \
+        '2\nn\n' \
+        '4\nn\n' \
+        '5\nn\n' \
+        '10\nn\n' \
+        '11\nn\n' \
+        '12\nn\n' \
+        '13\nn\n' \
+        '14\nn\n' \
+        '17\nn\n' \
+        '18\nn\n' \
+        '19\nn\n' \
+        '22\nn\n' \
+        '0\n' \
+        '{1:s}' \
+        '{2:d} {3:d}\n' \
+        '0'.format(project_name, str2, n_pressures, n_temperatures)
+    
+
+    p = Popen(werami_path, stdout=PIPE, stdin=PIPE, stderr=STDOUT)
+    stdout = p.communicate(input=stdin)[0]
+    print(stdout)
+    out = [s for s in stdout.split('\n') if "Output has been written to the" in s][0].split()[-1]
+    rename(out, outfile)
+    print('Output file renamed to {0:s}'.format(outfile))
+    print('Processing complete')
+
+    
+def read_2D_perplex_file(filename):
+    with open(filename, 'r') as f:
```

Comment:
this is an internal function, right? Maybe make this a member of the PerPlexMaterial or at least start with "_"?